Indexing/SubstitutionTree.cpp

```diff
@@ -72,10 +76,8 @@ SubstitutionTree::~SubstitutionTree()
   CALL("SubstitutionTree::~SubstitutionTree");
   ASS_EQ(_iteratorCnt,0);
 
-  for (unsigned i = 0; i<_nodes.size(); i++) {
-    if(_nodes[i]!=0) {
-      delete _nodes[i];
-    }
+  if (_root) {
```

Comment:
My understanding is that `delete nullptr;` has no effect, so this should just be `delete _root`.